plasmapy/particles/particle_class.py

```diff
@@ -2482,3 +2484,181 @@ def molecule(
 >>> def is_electron(particle: ParticleLike):
 ...     return particle == Particle("e-")
 """
+
+
+class Photon(AbstractParticle):
+    """
+    A class to represent a Photon.
+
+    Special case of an Abstract Particle with hardcoded mass,
+    charge, charge_number, spin and half_life to that of a
+    photon.
+
+    Parameters
+    ----------
+    quantity : ~astropy.units.Quantity, optional
+        The defining parameter of a Photon. It can be in the units of
+        Energy, Frequency, Momentum or Wavelength.
+
+    Raises
+    ------
+    ValueError
+        If the quantity is either not of allowed type or it's a negative quantity.
+
+    See Also
+    --------
+    ~plasmapy.particles.particle_class.Particle
+
+    Notes
+    -----
+    If the quantity is not specified, then the corresponding values of Energy,
+    Frequency, Momentum and Wavelength will be set to ``numpy.nan`` in
+    the appropriate units.
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.particles import Photon
+    >>> photon = Photon(5 * u.m)
+    >>> photon.wavelength
+    <Quantity 5. m>
+    >>> photon.energy
+    <Quantity 3.97289171e-26 J>
+    >>> photon.momentum
+    <Quantity 1.32521403e-34 J s / m>
+    >>> photon.frequency
+    <Quantity 59958491.6 1 / s>
+    >>> photon.spin
+    1
+    """
```

Comment:
```suggestion
    Examples
    --------
    We can use |Photon| to represent an
    :wikipedia:`Lyman-α photon <Lyman-alpha line>` for the
    transition of an electron from the :math:`n=2` orbital to the
    :math:`n=1` ground state in a hydrogen atom, where :math:`n` is
    the :wikipedia:`principal quantum number`.

    >>> from astropy import units as u
    >>> from plasmapy.particles import Photon
    >>> photon = Photon(121.567 * u.nm)
    >>> photon.wavelength
    <Quantity 1.21567e-07 m>
    >>> photon.energy
    <Quantity 1.63403379e-18 J>
    >>> photon.momentum
    <Quantity 5.45055003e-27 J s / m>
    >>> photon.frequency
    <Quantity 2.46606775e+15 1 / s>
    >>> photon.spin
    1
```
What would you think of using the example section to sneak in an educational example for a [Lyman-α photon](https://en.wikipedia.org/wiki/Lyman-alpha_line)? ☀️ 

I'm not sure that I'm using the `:wikipedia:` reStructuredText role correctly here, so we might need to tweak that.

Reply:
Ah I have studied this 5 years back but have no memory of it whatsoever but yeah I think it's a nice place to add it.